unlock-app/src/services/web3Service.js

```diff
@@ -319,12 +319,18 @@ export default class Web3Service extends EventEmitter {
   parseTransactionFromInput(transactionHash, contract, input, contractAddress) {
     const transactionType = this.getTransactionType(contract, input)
 
-    this.emit('transaction.updated', transactionHash, {
+    let update = {
       status: 'pending',
       type: transactionType,
       confirmations: 0,
       blockNumber: Number.MAX_SAFE_INTEGER, // Asign the largest block number for sorting purposes
-    })
+    }
+    // The lock a key is associated with is the same as the contract the key
+    // purchase transaction was sent to
+    if (transactionType === TRANSACTION_TYPES.KEY_PURCHASE) {
```

Comment:
I would put that logic in the `handler` called at line 352. The code here is not specific to any type of transaction while the handler actually exists for this specific purpose ;)

Reply:
Sure thing! Are these the relevant methods on the contract?
- `purchaseFor`
- `purchaseForFrom`